### PR TITLE
Add POSITIONS and BALANCES channels to Bitmex

### PR DESCRIPTION
**Pull request by KunduzB**
- Add POSITIONS channel and parser for it
- Add BALANCES channel and parser for it

Refer to the https://bit.ly/3leSEbN for data generalization

### CODE REVIEW

File: cryptofeed/exchanges/bitmex.py [hunk 1]
```diff
@@ -102,6 +107,48 @@ def parse_order(self, data):
...
+            position['leverage'] = data.get('leverage')
+        if data.get('unrealisedPnl') is not None:
+            position['unrealizedPnl'] = Decimal(data['unrealisedPnl'])
+        if data.get('currentQty') is not None:
+            position['contracts'] = data.get('currentQty')
+        if data.get('liquidationPrice') is not None:
+            position['liquidationPrice'] = Decimal(data['liquidationPrice'])
+        if data.get('markPrice') is not None:
+            position['markPrice'] = Decimal(data['markPrice'])
+        return position
```
Comment (boseok-bc) [inline cryptofeed/exchanges/bitmex.py]: Why does it decimalize some numbers and others are not?
How about converting all numbers to decimal for generalization?

File: cryptofeed/exchanges/bitmex.py [hunk 2]
```diff
@@ -581,6 +636,158 @@ async def _order(self, msg: dict, timestamp: float):
                 symbol = self.exchange_symbol_to_std_symbol(data['symbol'])
                 await self.callback(ORDER_INFO, feed=self.id, symbol=symbol, **new_info)
 
+    async def _position(self, msg: dict, timestamp: float):
+        """
+        {
```
Comment (boseok-bc) [inline cryptofeed/exchanges/bitmex.py:641]: just nitpick :) 
no `}`
indent contents

File: cryptofeed/exchanges/bitmex.py [hunk 3]
```diff
@@ -620,8 +831,13 @@ async def subscribe(self, conn: AsyncConnection):
         await self._authenticate(conn)
         chans = []
         for chan in self.subscription:
-            for pair in self.subscription[chan]:
-                chans.append(f"{chan}:{pair}")
+            print(chan)
+            if chan in Bitmex.channels_without_symbol:
+                print("hee")
```
Comment (boseok-bc) [inline cryptofeed/exchanges/bitmex.py]: Remove prints

**Review by boseok-bc — APPROVED** [review]
You can merge this PR after applying the comments 👍

**Comment by heartbeat31** [comment]
checked

**Comment by hotag-vero** [comment]
Checked

File: cryptofeed/exchanges/bitmex.py [hunk 4]
```diff
@@ -115,21 +115,21 @@ def parse_position(self, data):
         if data.get('symbol') is not None:
             position['symbol'] = self.exchange_symbol_to_std_symbol(data['symbol'])
         if data.get('initMargin') is not None:
-            position['initialMargin'] = data.get('initMargin')
+            position['initialMargin'] = Decimal(data.get('initMargin'))
         if data.get('maintMargin') is not None:
             position['maintenanceMargin'] = Decimal(data['maintMargin'])
         if data.get('avgEntryPrice') is not None:
-            position['entryPrice'] = data.get('avgEntryPrice')
+            position['entryPrice'] = Decimal(data.get('avgEntryPrice'))
```
Comment (hotag-vero) [inline cryptofeed/exchanges/bitmex.py:122]: How about 'avgEntryPrice' to distinguish from 'entryPrice' of each execution?

File: cryptofeed/exchanges/bitmex.py [hunk 5]
```diff
@@ -581,6 +636,158 @@ async def _order(self, msg: dict, timestamp: float):
...
+        'realisedPnl': -1385, 
+        'unrealisedGrossPnl': 11955, 
+        'longBankrupt': 0, 
+        'shortBankrupt': 0, 
+        'taxBase': 0, 
+        'indicativeTaxRate': None, 
+        'indicativeTax': 0, 
+        'unrealisedTax': 0, 
+        'unrealisedPnl': 11955, 
+        'unrealisedPnlPcnt': Decimal('0.0313'),
```
Comment (hotag-vero) [inline cryptofeed/exchanges/bitmex.py:715]: 'Pcnt' --> 'Pct' ?

**Pull request merged**